tox.ini

```diff
@@ -7,6 +7,10 @@ tox_pip_extensions_ext_venv_update = true
 
 [testenv]
 passenv = SSH_AUTH_SOCK
+# NO_PROXY is needed to call requests API within a forked process
+# when using macOS and python version 3.6/3.7
+setenv =
+    NO_PROXY = '*'
```

Comment:
Is this only a testing issue or do we need to somehow enable this in a production mode as well?

Reply:
Can we verify that this was existing in some sort of integration test / brute force testing. I would just like to ensure we are not introducing any new issues to the repository.

Reply:
If this is caused by `requests`, could it be the case that a newer version doesn't have this issue anymore? I see we're using `2.26.0`, while the latest one is `2.28.1`.

Reply:
Seems like [requests v2.28.0](https://requests.readthedocs.io/en/latest/community/updates/#id2) dropped support for python 3.6, I tried `requests` v2.27.1 with no luck. 

It's worth noting that the issue can be fixed if we use `spawn` context for multiprocessing but I think that would come at a performance cost.

Reply:
Right. So python 3.8+ multiprocessing library has switched to spawn where before it was fork. This is an issue since spawn does not share any memory between the parent and child processes. So we default the multiprocessing to fork when creating a pool of processes

Reply:
Can we try and use `v2.28.1`. Python 3.6 is end of life. We can consider dropping support for it.

Reply:
I tried `v2.28.1`, still the same problem. I think this is more of a python issue than requests, the default context was changed to spawn because of crashes similar to this. 

Since we are already using default multiprocessing (spawn) for python 3.8+ on macOS, what do you think of checking the current OS and if it's macOS we change to spawn?

Reply:
Sorry, that was the initial attempt to fix it. We decided to try and stay away from being OS specific so instead of changing the default multiprocessing, we instead passed the shared settings to child processes from the parent. So in python 3.8+ we are using spawn which is the default.